cypress/support/login.ts

```diff
@@ -0,0 +1,86 @@
+/*
+Copyright 2022 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/// <reference types="cypress" />
+
+import Chainable = Cypress.Chainable;
+import { SynapseInstance } from "../plugins/synapsedocker";
+
+export interface IUserCredentials {
+    accessToken: string;
+    userId: string;
+    deviceId: string;
+    password: string;
+    homeServer: string;
+}
+
+declare global {
+    // eslint-disable-next-line @typescript-eslint/no-namespace
+    namespace Cypress {
+        interface Chainable {
+            /**
+             * Generates a test user and instantiates an Element session with that user.
+             * @param synapse the synapse returned by startSynapse
+             * @param displayName the displayName to give the test user
+             */
+            initTestUser(synapse: SynapseInstance, displayName: string): Chainable<IUserCredentials>;
+        }
+    }
+}
+
+Cypress.Commands.add("initTestUser", (synapse: SynapseInstance, displayName: string): Chainable<IUserCredentials> => {
+    const username = Cypress._.uniqueId("userId_");
+    const password = Cypress._.uniqueId("password_");
```

Comment:
This is fun - is this a cypress internal or something and what's it doing?

Reply:
Its a re-exported lodash